hathor/transaction/storage/transaction_storage.py

```diff
@@ -1019,7 +1024,12 @@ def _run_topological_sort_dfs(self, root: BaseTransaction, visited: Dict[bytes,
 
     def add_to_indexes(self, tx: BaseTransaction) -> None:
         if not self.with_index:
-            raise NotImplementedError
```

Comment:
I'm wondering whether we should add a unit test that would fail before this fix.

Reply:
Hmmm. I'll do that I think it should be simple to make.

Reply:
I didn't actually create a test, but I made the situation appear in the tests, and added a protection to make sure the cache storage is set up correctly.